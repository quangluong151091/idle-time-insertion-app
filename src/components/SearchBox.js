import React from "react";

let nests = [
  {
    name: "29484001",
    img: "/pics/nest/29484001PO.png",
    link: '/29484001',
  },
  {
    name: "29484002",
    img: "/pics/nest/29484002PO.png",
    link: '/29484002',
  },
];

class SearchBar extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      searchString: "",
      nests: []
    };
    this.handleChange = this.handleChange.bind(this);
  }

  componentDidMount() {
    this.setState({
      nests: nests
    });
    this.refs.search.focus();
  }
  handleChange() {
    this.setState({
      searchString: this.refs.search.value
    });
  }
  render() {
    let _nests = this.state.nests;
    let search = this.state.searchString.trim();

    if (search.length > 0) {
      _nests = _nests.filter(function (nests) {
        return nests.name.match(search);
      });
    }
    return (
      <div className="search-box">
        <input
          id="search"
          className="search-txt"
          type="text"
          placeholder="Search..."
          value={this.state.searchString}
          ref="search"
          onChange={this.handleChange} />
        <a href="#searchbox" className="search-btn"><i className="fas fa-search"></i></a>
        <div id="searchlist" className='search-result' hidden>
          <ul>
            {_nests.map(l => {
              return (
                <a href={l.link} key={l.name}>
                  <li>
                    <img src={l.img} width='150px' alt="nestImg" />
                    <span className='text-list'>{l.name}</span>
                  </li>
                </a>
              );
            })}
          </ul>
        </div>
      </div>
    )
  }
}

export default SearchBar;