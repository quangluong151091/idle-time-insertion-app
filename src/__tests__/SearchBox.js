import React from 'react';
import { render, mount } from 'enzyme';

import SearchBox from '../components/SearchBox';

describe('SearchBox', () => {
  it('should render correctly', () => {

    const component = render(<SearchBox />);

    expect(component).toMatchSnapshot();
  });
  it('checks if handleChange method works correctly', () => {
    const wrapper = mount(<SearchBox />);

    const searchInput = wrapper.ref("search");
    searchInput.value = "123456";
    wrapper.find("#search").simulate('change');
    expect(wrapper.instance().state.searchString).toEqual('123456');
  });
});