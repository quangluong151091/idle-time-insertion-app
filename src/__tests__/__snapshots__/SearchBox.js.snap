// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBox should render correctly 1`] = `
<div
  class="search-box"
>
  <input
    class="search-txt"
    id="search"
    placeholder="Search..."
    type="text"
    value=""
  />
  <a
    class="search-btn"
    href="#searchbox"
  >
    <i
      class="fas fa-search"
    />
  </a>
  <div
    class="search-result"
    hidden=""
    id="searchlist"
  >
    <ul />
  </div>
</div>
`;
